doc/classes/String.xml

```diff
@@ -181,6 +181,25 @@
 			<argument index="1" name="placeholder" type="String" default="&quot;{_}&quot;" />
 			<description>
 				Formats the string by replacing all occurrences of [code]placeholder[/code] with [code]values[/code].
+				[b]Note:[/b] There must be one and only one [code]_[/code] in the [code]placeholder[/code].
+				[codeblocks]
+				[gdscript]
+				# Use the default placeholder, and a two-dimensional array.
+				var str1 = "name will be replaced to {name}, 0 maps {0}, strength is {strength}"
+				# Then the result1 is "name will be replaced to RobotGuy, 0 maps godot, strength is 9000.91".
+				var result1 = str1.format([["name", "RobotGuy"], [0, "godot"], ["strength", 9000.91]])
+
+				# Use a custom placeholder string "$_", and a dictionary.
+				var str2 = "red=$red green=$green blue=$blue alpha=$alpha"
+				# Then the result2 is "red=0.1 green=0.3 blue=0.5 dict={hello:godoter}".
+				var result2 = str2.format({"red": 0.1, "green": 0.3, "blue": 0.5, "dict": {"hello": "godoter"}}, "$_")
```

Comment:
Wait, this appears to be incorrect. The actual result is `red=0.1 green=0.3 blue=0.5 alpha=$alpha`

Reply:
Alright, but what's the point of the dictionary now? :stuck_out_tongue:

Reply:
> Wait, this appears to be incorrect. The result from this is `red=0.1 green=0.3 blue=0.5 alpha=$alpha`.

Yeah, it is, when transcribing, I want to show the possible situation, that is, there is no need for an exact match.

Reply:
> Alright, but what's the point of the dictionary now? stuck_out_tongue

Matching may not be done all at once. There may be some more complicated cases here, but this is just a demonstration of the usage of this method `format`, not the `Dictionary`.

Reply:
I still don't quite get it. Is the point of the example to show it **not** formatting "$alpha"? Because if so, a simpler example than a dictionary would be better.

Reply:
> I still don't quite get it. Is the point of the example to show it **not** formatting "$alpha"? Because if so, a simpler example than a dictionary would be better.

For a better example : 

`... company id is {company_id}, named {company_name}, ...`

If you just want to get a brief company information, and already have a dictionary like this:

`company_info =  {"company_id": 001, "company_name": "A", "departments":{"name":"a", "director":"Tom", "members":{...}}}`

You can just use `format` without having to extract the dictionary.

Reply:
I see that. I just don't understand the point of the `"dict": {"hello": "godoter"}` dictionary.

Reply:
> I see that. I just don't understand the point of the `"dict": {"hello": "godoter"}` dictionary.

It's just that I didn't know what to write at that time, I just wrote it casually.

Just want to tell the reader that nested dictionaries can be used in the function.

Reply:
> there is no need for an exact match.

There is no need for an exact match.

Reply:
> Just want to tell the reader that nested dictionaries can be used in the function. There is no need for an exact match.

Then maybe it could be explicitly written in the comment that the dictionary doesn't need to match it exactly. It would both explain it better and give a purpose to that sub-dictionary.